src/store/modules/authentication.ts

```diff
@@ -76,10 +93,19 @@ const AuthenticationModule: Module<AuthenticationState, {}> = {
       // TODO: You probably want put this into the `state`. -Niko
       // Im not sure how this should be done, so I did it as I know.
       context.commit('SET_AUTHENTICATION_TOKEN', extended)
+      return extended
+    },
+    [AuthenticationActions.STORE_TOKEN] (context: AuthenticationActionContext, token: string) {
+      localStorage.setItem(JWT_KEY, token)
     },
-    async [AuthenticationActions.LOGIN_WG] (context: AuthenticationActionContext, region: string) {
+    [AuthenticationActions.LOGIN_WG] (context: AuthenticationActionContext, region: JWTRegion) {
       // TODO: This is just placeholder logic. Please check it works. -Niko
-      location.replace(process.env.VUE_APP_MS_AUTH + `/connect/wargaming/${region}/${window.location.href}`)
+      const returnUrl = process.env.VUE_APP_MS_AUTH + `/connect/wargaming/${regionDomain(region)}/${window.location.href}`
```

Comment:
You cant expect all regions to be available for login. Please use authentication microservice index to fetch available providers.